### DISCUSSION TITLE
How to deploy NiceGUI application to a website?

### DISCUSSION THREAD

**Discussion by 200729** [zauberzeug/nicegui#20] [Q&A] [answered]
How to deploy nicegui application to the website?
Can nicegui be integrated with other web framework e.g. Flask?

**Comment by rodja** [comment, marked as answer]
Good question. I've added the new section "Deployment" to our README: https://github.com/zauberzeug/nicegui/blob/main/README.md#deployment
Does that help you to get started? Or is something crucial missing?

About your question for integration with other web frameworks like Flask:
NiceGUI is based on [JustPy](https://justpy.io/) which is based on the ASGI framework [Starlette](https://www.starlette.io/) and the ASGI webserver [Uvicorn](https://www.uvicorn.org/). This is a fundamentally different stack than Flask. In NiceGUI you can define Starlette routes with any data response you like (see https://nicegui.io/#route). With this you can add RESTful endpoints, provide file downloads and similar.
If you want to run different web frameworks side by side, I suggest you place them behind a reverse proxy like [Traefik](https://doc.traefik.io/traefik/) or [NGINX](https://www.nginx.com/).

**Reply by rodja** [reply]
We moved and expanded the deployment documentation to https://nicegui.io/documentation/section_configuration_deployment#server_hosting

**Comment by TeeFiX42** [comment]
Hi, I get a 404 on the last url. - https://nicegui.io/reference#server_hosting
Can you verify, pls?

**Reply by rodja** [reply]
Oh thanks for the feedback. I updated the url in the commend above.
